main/src/ca/uwaterloo/flix/language/phase/Safety.scala

```diff
@@ -492,6 +495,42 @@ object Safety {
       }
   }
 
+  /**
+    * Converts associated types in `tpe0` to concrete types.
```

Comment:
Optionally returns ...

Reply:
The implementation is perhaps surprising: You fail if the type contains one assoc type that cannot be resolved, yet you still try to resolve all assoc types. 

Would it not be better to make it a Type => Type function. With the spec that is replaces all assoc types that can trivially be resolved. This seems like a more meaningful definition to me. It also avoids the monadic stuff, while being generally more useful.